activerecord/lib/active_record/connection_adapters/pool_config.rb

```diff
@@ -41,11 +40,33 @@ def server_version(connection)
         @server_version || synchronize { @server_version ||= connection.get_database_version }
       end
 
+      def connection_class=(connection_class)
+        case connection_class
+        when ConnectionHandler::StringConnectionName
+          @connection_class_name = connection_class
+        else
+          @connection_class_name = connection_class.name
+        end
+        @connection_class_primary_p = connection_class.primary_class?
+      end
+
+      def connection_class
+        case @connection_class_name
+        when ConnectionHandler::StringConnectionName
+          @connection_class_name
+        else
+          @connection_class_name.constantize
+        end
```

Comment:
Think method is quite hot. `constantize` isn't that bad, but I think I'd prefer if we could avoid it in the happy path.

I suspect 99% of the time `connection_class` is `ActiveRecord::Base` which shouldn't be reloadable, might be worth having a fastpath for that.